src/core/space_interaction_API.jl

```diff
@@ -304,8 +304,7 @@ nearby_agents(a, model, r = 1; kwargs...) =
 
 """
     random_nearby_id(agent, model::ABM, r = 1, f = nothing, alloc = false; kwargs...) → id
-Return the `id` of a random agent near the position of the given `agent` using an optimized
-algorithm from [Reservoir sampling](https://en.wikipedia.org/wiki/Reservoir_sampling#An_optimal_algorithm).
+Return the `id` of a random agent near the position of the given `agent`. 
```

Comment:
It's fine to not say too many details, but i think it is still improtant to say that this function does not require collecting all the agents and randomly sampling one.

Reply:
reintroduced in a similar form